lib/release.js

```diff
@@ -0,0 +1,437 @@
+        arr[idx] = `#define NODE_MINOR_VERSION ${versionComponents.minor}`;
+      } else if (line.includes('#define NODE_PATCH_VERSION')) {
+        arr[idx] = `#define NODE_PATCH_VERSION ${versionComponents.patch}`;
+      } else if (line.includes('#define NODE_VERSION_IS_RELEASE')) {
+        arr[idx] = '#define NODE_VERSION_IS_RELEASE 1';
+      } else if (line.includes('#define NODE_VERSION_IS_LTS')) {
+        this.isLTS = arr[idx].split(' ')[2] === '1';
+        this.ltsCodename = arr[idx + 1].split(' ')[2];
+      }
+    });
+
+    await fs.writeFile(filePath, arr.join('\n'));
+  }
+
+  updateNodeModuleVersion(runtime, variant, versions) {
+    const nmvFilePath = path.resolve('doc', 'abi_version_registry.json');
+    const nmvArray = readJson(nmvFilePath).NODE_MODULE_VERSION;
+
+    const latestNMV = nmvArray[0];
+    const modules = latestNMV.modules + 1;
+    nmvArray.unshift({ modules, runtime, variant, versions });
+
+    writeJson(nmvFilePath, { NODE_MODULE_VERSION: nmvArray });
+  }
+
+  async createReleaseCommit() {
+    const { cli, isLTS, newVersion, isSecurityRelease, date } = this;
+
+    const releaseType = isLTS ? 'LTS' : 'Current';
+    const messageTitle = `${date} Version ${newVersion} (${releaseType})`;
+
+    const messageBody = [];
+    if (isSecurityRelease) {
+      messageBody.push('This is a security release.\n\n');
+    }
+
+    const notableChanges = this.getBranchDiff(true);
+    messageBody.push('Notable changes:\n\n');
+    messageBody.push(notableChanges);
+
+    // Create commit and then allow releaser to amend.
+    runSync('git', ['add', '.']);
+    runSync('git', [
+      'commit',
+      '-m',
+      messageTitle,
+      '-m',
+      messageBody.join('')
+    ]);
+
+    cli.log(`${messageTitle}\n\n${messageBody.join('')}`);
+    const useMessage = await cli.prompt('Continue with this commit message?', true);
+    return useMessage;
+  }
+
+  getBranchDiff(onlyNotableChanges = false) {
+    const {
+      versionComponents,
+      stagingBranch,
+      upstream,
+      newVersion,
+      isLTS
+    } = this;
+
+    let branchDiffOptions;
+    if (onlyNotableChanges) {
+      const proposalBranch = `v${newVersion}-proposal`;
+      const releaseBranch = `v${versionComponents.major}.x`;
+
+      const notableLabels = [
+        'notable-change',
+        'semver-minor'
+      ];
+
+      branchDiffOptions = [
+        'branch-diff',
+        `${upstream}/${releaseBranch}`,
+        proposalBranch,
+        `--require-label=${notableLabels.join(',')}`,
+        '-format=simple'
+      ];
+    } else {
+      const excludeLabels = [
+        'semver-major',
+        `dont-land-on-v${versionComponents.major}.x`,
+        `backport-requested-v${versionComponents.major}.x`,
+        `backported-to-v${versionComponents.major}.x`,
+        `backport-blocked-v${versionComponents.major}.x`
+      ];
+
+      if (isLTS) {
+        excludeLabels.push('semver-minor');
```

Comment:
there are times where LTS includes semver-minor fwiw so this check should likely be `isLTS && !semverMinor`